src/Core/Grid/Definition/Factory/CartRuleGridDefinitionFactory.php

```diff
@@ -66,9 +67,10 @@ final class CartRuleGridDefinitionFactory extends AbstractGridDefinitionFactory
 
     public function __construct(
         HookDispatcherInterface $hookDispatcher,
+        TranslatorInterface $translator,
         string $contextDateFormat
     ) {
-        parent::__construct($hookDispatcher);
+        parent::__construct($hookDispatcher, $translator);
```

Comment:
I'm sceptical about adding optional dependencies inside the constructor. As it can sometimes be useless like here for example.
Perhaps adding a translatable trait, would avoid injecting it everywhere. Avoiding the same error we made for 

WDYT ?

Reply:
the trans method must be present in AbstractGridDefinition to avoid BC breaks, but as you suggested I tried to get back the previous implementation of TranslatorAwareTrait and calls: "setTranslator", but had no luck, it required to add "calls: setTranslator to every implementation, I couldn't find a way how to go around it :thinking: